custom_model_runner/datarobot_drum/resource/predict_mixin.py

```diff
@@ -74,45 +66,28 @@ def _predict_or_transform(self, logger=None):
             )
             if logger is not None:
                 logger.error(wrong_key_error_message)
+            raise ValueError(wrong_key_error_message)
+        return binary_data, mimetype, charset
+
+    def _predict(self, logger=None):
+        response_status = HTTP_200_OK
+
+        file_key = "X"
```

Comment:
As long as request is available in every method, you don't need to keep file_key and filestorage in this method and pass into _fetch_data. You can just access request from_fetch_data.
Also maybe rename _fetch_data to _fetch_Data_from_request

Reply:
and you can make fetch data to return tuple of two for predict and tuple of three for transform

Or tuple of five